common-npm-packages/azure-arm-rest-v2/azure-arm-storage.ts

```diff
@@ -217,6 +218,33 @@ export class StorageAccounts {
         return "";
     }
 
+    private async getStorageAccountsByUri(uri: string, options?: any): Promise<Model.StorageAccount[]> {
+        const request = new webClient.WebRequest();
+        request.method = 'GET';
+        request.headers = this.client.setCustomHeaders(options);
+        request.uri = uri;
+
+        const storageAccounts: Model.StorageAccount[] = [];
+
+        const response = await this.client.beginRequest(request);
+        if (response.statusCode !== 200) {
+            throw azureServiceClientBase.ToError(response);
+        }
+
+        if (response.body.nextLink) {
+            const nextResult = await this.client.accumulateResultFromPagedResult(response.body.nextLink);
+            if (nextResult.error) {
+                throw nextResult.error;
+            }
+
+            storageAccounts.push(...nextResult.result);
+        }
+
+        storageAccounts.push(...response.body.value);
```

Comment:
Maybe you can move it before the `nextLink` `if control` to get at least first page of results first.
So we don't need to fetch `nextLink` if we find name. (please check optimization idea)

Reply:
Fixed, thanks!